src/query/sql/src/planner/binder/project.rs

```diff
@@ -290,3 +308,79 @@ impl<'a> Binder {
         Ok(output)
     }
 }
+
+// Pre-check exclude_col is legal
+fn precheck_exclude_cols(
+    input_context: &BindContext,
+    exclude_cols: &HashSet<String>,
+    db_name: Option<&Identifier>,
+    table_name: Option<&Identifier>,
+) -> Result<()> {
+    let all_columns_bind = input_context.all_column_bindings();
+    let mut qualified_cols_name: HashMap<String, u8> = HashMap::new();
+
+    fn fill_qualified_cols(
+        qualified_cols_name: &mut HashMap<String, u8>,
+        column_bind: &ColumnBinding,
+    ) {
+        let col_name = column_bind.column_name.clone();
+        if qualified_cols_name.contains_key(col_name.as_str()) {
```

Comment:
we can return `ambiguous column name '{exclude_col}'` here, without contiune check.

Reply:
> we can return `ambiguous column name '{exclude_col}'` here, without contiune check.

Done